cmd/kube-proxy/app/server.go

```diff
@@ -557,7 +557,7 @@ func (s *ProxyServer) Run() error {
 		if encounteredError {
 			return errors.New("encountered an error while tearing down rules")
 		}
-		return nil
+		return errors.New("cleaup and exit")
```

Comment:
Agree with @vllry, this seems like a bit of a code smell, we're effectively saying that nil errors are invalid. I personally think we should fix/refactor properly in this PR. I'm wondering if it would be cleaner if we pulled the clean up & exit logic out of `Run()` and ran the clean up check prior, that would reduce some of the complexity with passing errors through channels.

Reply:
That is a good way!
It can run clean up & exit logic before run proxy server if success then os exit else return the error message.
I will refactor in this PR.
https://github.com/kubernetes/kubernetes/blob/3b12266c06709f5a5ada9fd9605839ee8a796ef6/cmd/kube-proxy/app/server.go#L321-L325

Reply:
Yeah that seems like a better approach to me, thoughts @vllry?

Reply:
+1 from me